lib/Workflow.pm

```diff
@@ -120,8 +120,10 @@ sub execute_action {
                 );
 
             if ( $wf_state->may_stop() ) {
-                $action_name =
-                    eval { $wf_state->get_autorun_action_name($self); };
+                try {
+                    $action_name = $wf_state->get_autorun_action_name($self);
+                }
+                catch { }
```

Comment:
Do we need error handling here?

Reply:
Good question. The current situation is used to handle https://github.com/jonasbn/perl-workflow/blob/master/lib/Workflow/State.pm#L144-L147, which feels the need to throw an error instead of returning undef or an empty list when *no* autorun actions are available. It also throws an error (which is ignored!) when *multiple* autorun actions are available. This is likely a situation that actually should be propagated to callers.

How about changing the semantics of `get_autorun_action_name()` in 2.0 to return undef for none, a string for one available and an error for multiple available? Then we should not even need to catch an error here.

Reply:
I guess I can summarize my answer as "no" or as "we can't at the moment" (the latter is obviously something we can work on...)